p2p/gater.go

```diff
@@ -43,3 +78,38 @@ func (*gater) InterceptSecured(_ network.Direction, _ peer.ID, _ network.ConnMul
 func (*gater) InterceptUpgraded(_ network.Conn) (allow bool, reason control.DisconnectReason) {
 	return true, 0
 }
+
+func (g *gater) isBlocked(m multiaddr.Multiaddr) bool {
+	blocked := false
+	multiaddr.ForEach(m, func(c multiaddr.Component) bool {
+		switch c.Protocol().Code {
+		case multiaddr.P_IP4:
+			blocked = inAddrRange(net.IP(c.RawValue()), g.ip4blocklist)
+		case multiaddr.P_IP6:
+			blocked = inAddrRange(net.IP(c.RawValue()), g.ip6blocklist)
+		}
+		return false
```

Comment:
there is no need to inspect all components?

Reply:
i think there is an assumption that ip4/ip6 is a first component, and i just copied this code. but i changed it to iterate just in case